foundation/gateway/internal/provider/regionalprovider/network.go

```diff
@@ -56,9 +61,25 @@ func NewNetworkController(logger *slog.Logger, cfg *rest.Config) (*NetworkContro
 		return nil, fmt.Errorf("failed to create kubeclient: %w", err)
 	}
 
+	convert := func(u unstructured.Unstructured) (*regional.NetworkSKUDomain, error) {
+		var crdNetworkSKU skuv1.NetworkSKU
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &crdNetworkSKU); err != nil {
+			return &regional.NetworkSKUDomain{}, err
+		}
+		return regional.FromCRToNetworkSKUDomain(crdNetworkSKU), nil
+	}
+
+	networkSKUAdapter := kubernetes.NewAdapter(
+		client.Client,
+		skuv1.NetworkSKUGVR,
+		logger,
+		convert,
+	)
```

Comment:
adapters should be handed over to controllers, to make dependency injection possible. A controller usually only imports model and port.